examples/mxnet/gc-mc/train.py

```diff
@@ -0,0 +1,313 @@
+                                             prefix='gen_rating')
+            else:
+                self.gen_ratings = InnerProductLayer(prefix='gen_rating')
+
+
+    def forward(self, user_fea, movie_fea, rating_node_pairs):
+        user_out, movie_out = self.encoder(user_fea, movie_fea)
+        # Generate the predicted ratings
+        rating_user_fea = mx.nd.take(user_out, rating_node_pairs[0])
+        rating_item_fea = mx.nd.take(movie_out, rating_node_pairs[1])
+        pred_ratings = self.gen_ratings(rating_user_fea, rating_item_fea)
+
+        return pred_ratings
+
+
+
+def evaluate(args, net, feature_dict, dataset, segment='valid'):
+    rating_mean = dataset.train_rating_values.mean()
+    rating_std = dataset.train_rating_values.std()
+    possible_rating_values = dataset.possible_rating_values
+    nd_possible_rating_values = mx.nd.array(possible_rating_values, ctx=args.ctx, dtype=np.float32)
+
+    if segment == "valid":
+        rating_pairs = dataset.valid_rating_pairs
+        rating_values = dataset.valid_rating_values
+    elif segment == "test":
+        rating_pairs = dataset.test_rating_pairs
+        rating_values = dataset.test_rating_values
+    else:
+        raise NotImplementedError
+    user_input = mx.nd.array(feature_dict["user"], ctx=args.ctx, dtype=np.float32)
+    movie_input = mx.nd.array(feature_dict["movie"], ctx=args.ctx, dtype=np.float32)
+
+    rating_pairs = mx.nd.array(rating_pairs, ctx=args.ctx, dtype=np.int64)
+    rating_values = mx.nd.array(rating_values, ctx=args.ctx, dtype=np.float32)
+
+    # Evaluate RMSE
+    pred_ratings = net(user_input, movie_input, rating_pairs)
+    if args.gen_r_use_classification:
+        real_pred_ratings = (mx.nd.softmax(pred_ratings, axis=1) *
+                             nd_possible_rating_values.reshape((1, -1))).sum(axis=1)
+        rmse = mx.nd.square(real_pred_ratings - rating_values).mean().asscalar()
+    else:
+        rmse = mx.nd.square(mx.nd.clip(pred_ratings.reshape((-1,)) * rating_std + rating_mean,
+                                       possible_rating_values.min(),
+                                       possible_rating_values.max()) - rating_values).mean().asscalar()
+    rmse  = np.sqrt(rmse)
+    return rmse
+
+def train(args):
+    dataset, feature_dict = load_dataset(args)
+    print("Loading data finished ...\n")
+
+    ### prepare data
+    possible_rating_values = dataset.possible_rating_values
+    nd_possible_rating_values = mx.nd.array(possible_rating_values, ctx=args.ctx, dtype=np.float32)
+    train_rating_pairs = mx.nd.array(dataset.train_rating_pairs, ctx=args.ctx, dtype=np.int64)
+    train_gt_ratings = mx.nd.array(dataset.train_rating_values, ctx=args.ctx, dtype=np.float32)
+    rating_mean = dataset.train_rating_values.mean()
+    rating_std = dataset.train_rating_values.std()
+
+    print("Start preparing graph ...")
+    uv_train_graph = dataset.uv_train_graph
+    vu_train_graph = dataset.vu_train_graph
```

Comment:
should we remove the training edges from these two graphs? It seems we can use edge_subgraph to remove the training edges from the original graph.